index.js

```diff
@@ -12,10 +12,35 @@ db.prepare("CREATE TABLE IF NOT EXISTS streetnumbers (id INTEGER UNIQUE PRIMARY
 var streets = d3.csvParse(fs.readFileSync('./data/streets.csv','utf8'))
 
 streets.forEach((s,i)=>{
-  var plz = parseInt(s.plz)
-  if(isNaN(plz)){
-    plz = 0
+  
+  var plz;
+  
+  if ((s.plz).includes("|")) {
+   
+    var arr = s.plz.split("|");
+
+    arr.forEach((el) => {
+      plz = parseInt(el);
+      
+      if (isNaN(plz)) {
+      
+        plz = 0;
+      
+      } 
+        var params = [s.street, plz];
+        db.prepare('INSERT INTO streets (street, plz) VALUES (?,?)').run(params);
+      
+    });
+
+  } else {
+
+    plz = parseInt(s.plz)
+    
+    if (isNaN(plz)) {
+      plz = 0;
+    }
   }
+
   var params = [s.street, plz]
```

Comment:
Move this INSERT inside the else statement after line 41. Otherwise we will get duplicates for the ones that include "|".

Reply:
Together with the params, of course.